src/main/java/org/tomitribe/auth/signatures/Signature.java

```diff
@@ -83,32 +100,48 @@
      * Construct a signature configuration instance with the specified keyId, algorithm and HTTP headers.
      * 
      * @param keyId An opaque string that the server can use to look up the component they need to validate the signature.
+     * @param signingAlgorithm An identifier for the HTTP Signature algorithm.
+     *  This should be "hs2019" except for legacy applications that use an older version of the draft HTTP signature specification.
+     * @param algorithm The detailed algorithm used to sign the message.
      * @param parameterSpec optional cryptographic parameters for the signature.
-     * @param algorithm The algorithm used to sign the message.
      * @param headers The list of HTTP headers that will be used in the signature.
      */
-    public Signature(final String keyId, final String algorithm, final AlgorithmParameterSpec parameterSpec, final String... headers) {
-        this(keyId, getAlgorithm(algorithm), parameterSpec, null, Arrays.asList(headers));
+    public Signature(final String keyId, final String signingAlgorithm, final String algorithm, final AlgorithmParameterSpec parameterSpec, final List<String> headers) {
```

Comment:
I wonder if we can keep the old constructors and mark them with `@Deprecated` in order to maintain some backwards compatibility for other clients?

Reply:
I suspect String... headers makes keeping the old constructors not possible.